enhancements/ingress/global-options-enable-hsts.md

```diff
@@ -0,0 +1,252 @@
+    - `max-age` decrements over time
+- `includeSubDomains`: if present, applies to any subdomains of the host's domain name
+- `preload`: if present, tells the client to include this host in its host preload list so that 
+it never needs to do an initial load to get the HSTS header (note that this is not defined in RFC 6797 
+and is therefore client implementation-dependent)
+
+HSTS is currently implemented as a pattern in the HAProxy template and applied to `edge` and 
+`reencrypt` routes that match the `haproxy.router.openshift.io/hsts_header` annotation:
+````
+{{- /* hsts header in response: */}}
+ {{- $hstsOptionalTokenPattern := `(?:includeSubDomains|preload)` }}
+ {{- $hstsPattern := printf `(?:%[1]s[;])*max-age=(?:\d+|"\d+")(?:[;]%[1]s)*`  $hstsOptionalTokenPattern -}}
+...
+{{- if matchValues (print $cfg.TLSTermination) "edge" "reencrypt" }}
+    {{- with $hsts := firstMatch $hstsPattern (index $cfg.Annotations "haproxy.router.openshift.io/hsts_header") }}
+  http-response set-header Strict-Transport-Security {{$hsts}}
+    {{- end }}{{/* hsts header */}}
+  {{- end }}{{/* is "edge" or "reencrypt" */}}
+````
+#### Enable HSTS for all routes globally
+To add the header to all HTTP responses from an `edge` or `reencrypt` route, add a new router environment 
+variable called `ROUTER_DEFAULT_HSTS`, configured by the administrator in the Ingress Operator.  `ROUTER_DEFAULT_HSTS` is detected in the HAProxy configuration file processing, and can be overridden
+by any individual route's annotation.  Global HSTS does not override an existing route annotation for 
+`haproxy.router.openshift.io/hsts_header`
+
+`ROUTER_DEFAULT_HSTS` should be blank if it is not needed, otherwise it should define `max-age` and optionally specify
+`includeSubDomains` and `preload`.  An example of a properly formatted `ROUTER_DEFAULT_HSTS` that expires after one year
+and includes subdomains looks like:
+````
+max-age=31536000; includeSubDomains
+````
+With the addition of processing for `ROUTER_DEFAULT_HSTS`, the code snippet above becomes:
+````bash
+{{- $router_global_hsts := env "ROUTER_DEFAULT_HSTS" "" }}
+...
+{{- if matchValues (print $cfg.TLSTermination) "edge" "reencrypt" }}
+  {{- /* First check the environment to see if the header is set globally */}}
+  {{- if (ne "" $router_global_hsts) }} 
+    http-response set-header Strict-Transport-Security {{$router_global_hsts}}
+  {{- end }}{{/* global_hsts */}}
+  {{- /* Override if the annotation exists.  Set-header replaces any existing header */ }}
+  {{- with $hsts := firstMatch $hstsPattern (index $cfg.Annotations "haproxy.router.openshift.io/hsts_header") }}
+    http-response set-header Strict-Transport-Security {{$hsts}}
+  {{- end }}{{/* hsts header override per-route */}}
+{{- end }}{{/* is "edge" or "reencrypt" */}}
+````
+
+#### Enable HSTS on a node-type basis (infra-only, worker-only, or both)
+(TBD) Clarification has been requested on this requirement.
```

Comment:
Being able to enable HSTS on a per ingress controller basis may satisfy the customer's desire to be able to apply different HSTS settings for particular subsets of routes (ie routes that target workloads running on infra nodes, as long as all of those routes are served by the same ingress controller). Based on the [NE-310](https://issues.redhat.com/browse/NE-310) epic description, I get the impression that the customer wants to be able to apply unique HSTS settings for the default ingress controller, but we should try to clarify what exactly their requirements are.

Reply:
In this case, they clarified that they would like it on a per-router basis.  I updated the doc for this case, can you ptal?